test/mechanics/integrands/HistoryData.cpp

```diff
@@ -0,0 +1,397 @@
+        return std::exp(-delta_t / mT_KC[index]);
+    }
+
+    //! @brief Calculates the algorithm specific parameter lambda
+    //! @param delta_t: Time increment
+    //! @param index: Index of the Kelvin Unit
+    //! @return Algorithm specific parameter lambda
+    double Lambda(double delta_t, unsigned int index) const
+    {
+        assert(index < mT_KC.rows());
+        return mT_KC[index] / delta_t * (1 - Beta(delta_t, index));
+    }
+
+    //! @brief Calculates the creep strain increment
+    //! @param hisData: History data object
+    //! @param delta_t: Time increment
+    //! @return Creep strain increment
+    EngineeringStrainPDE<1> DeltaCreep(const CreepHistoryData& hisData, double delta_t) const
+    {
+        EngineeringStrainPDE<1> deltaCreep;
+        deltaCreep[0] = 0.;
+        for (unsigned int i = 0; i < mE_KC.rows(); ++i)
+            deltaCreep[0] += (1. - Beta(delta_t, i)) * hisData.gamma[i];
+        return deltaCreep;
+    }
+};
+
+
+// %%% Mesh generation %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+MeshFem Mesh1D()
+{
+    MeshFem mesh;
+    const InterpolationSimple& interpolation = mesh.CreateInterpolation(InterpolationTrussLinear(1));
+    NodeSimple* nr = nullptr;
+    constexpr unsigned int numNodes = numElements + 1;
+    constexpr double elementLength = SpecimenLength / numElements;
+    for (unsigned int i = 0; i < numNodes; ++i)
+    {
+        NodeSimple& nl = mesh.Nodes.Add({i * elementLength});
+        if (i > 0)
+            mesh.Elements.Add({{{nl, *nr}, interpolation}});
+        nr = &nl;
+    }
+
+
+    return mesh;
+}
+
+
+// %%% Main %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+using namespace std::placeholders;
+
+BOOST_AUTO_TEST_CASE(History_Data)
+{
+    // Create mesh %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    MeshFem mesh = Mesh1D();
+    DofType displ("displacements", 1);
+    const auto& interpolation = mesh.CreateInterpolation(InterpolationTrussLinear(1));
+    AddDofInterpolation(&mesh, displ, interpolation);
+
+
+    // Create constraints %%%%%%%%%%%%%%%%%%%%%%%
+    ConstraintPde::Constraints constraints;
+    auto& nodeLeft = mesh.NodeAtCoordinate(Eigen::VectorXd::Zero(1), displ);
+    constraints.Add(displ, ConstraintPde::Component(nodeLeft, {eDirection::X}));
+
+
+    // DOF numbering %%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    DofNumbering::DofInfo dofInfo = DofNumbering::Build(mesh.NodesTotal(displ), displ, constraints);
+
+
+    // Create law %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    constexpr double E = 40000;
+    const Vector2d E_KC(60000., 120000.);
+    const Vector2d D_KC(1., 2.);
+    CreepLaw creepLaw(E, E_KC, D_KC);
+
+
+    // Create integrand %%%%%%%%%%%%%%%%%%%%%%%%%
+    Integrands::MomentumBalance<1> momentumBalance(displ, creepLaw);
+    double delta_t = 0.1;
+    auto MomentumGradientF =
+            std::bind(&Integrands::MomentumBalance<1>::Gradient, momentumBalance, _1, _2, std::ref(delta_t));
```

Comment:
Why is here the reference to delta t? This is the kind of function call that follows the patter
Set(t), Gradient() instead of Gradient(t). When evaluating the gradient, I would like to see the time (or delta time) directly be added as a parameter (i.e. via a lambda)

Reply:
Well the answer is simple: 
Without the ref, every change in delta_t (dynamic timestepping) would not be passed to `MomentumGradientF`. So I would have to call std::bind again with the updated value.
The whole line
~~~cpp
 auto MomentumGradientF = std::bind(&Integrands::MomentumBalance<1>::Gradient, momentumBalance, _1, _2, std::ref(delta_t));
~~~

can also be written as

~~~cpp
auto MomentumGradientF = [&momentumBalance, &delta_t](cellData cd, cellIPData cipd){
    return momentumBalance.Gradient(cd,cipd,delta_t);
};
~~~

When we call `assembler.BuildVector` it expects a function which only has cellData and CellIPData as parameters, but no delta_t. So we need a lambda function or a std::bind to squeeze the gradient function of the law (which needs a delta_t) through the interface. As I said, if you remove the ref, you have to rebind the function every time delta_t changes. It is like removing the & in front of delta_t in the lambda capture.

Apart from that, I think this is not important to the history data itself. Maybe it should be discussed in another issue, if this stuff is going to be part of any core time integration scheme. In an example or test it shouldn't matter

Reply:
I thought about that as well. 
This is like a combination of [currying](https://en.wikipedia.org/wiki/Currying) and [partial application](https://en.wikipedia.org/wiki/Partial_application). Your gradient function takes 4 arguments. You want to transform that into a function, that takes `delta_t` and returns a function that takes `celldata, cellipdata` and partially applies the integrand. I solved that with a lambda that returns another lambda. It is a bit messy but can even be solved with templates and put somewhere in NuTo. 
I think I even put a snippet of that into our git.bam.de (no access atm...)

@vhirtham I think you are right that this may be a different topic. Ehr. But now I already typed it... :)